libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -258,15 +258,17 @@ public void updateDistanceWith(RouteProgress routeProgress) {
   }
 
   public void updateBannerInstructionsWith(BannerInstructions instructions) {
-    if (instructions == null || instructions.primary() == null) {
-      return;
+    if (instructions != null) {
+      updateBannerInstructions(instructions.primary(),
```

Comment:
Is it 🆗 passing `null` here `instructions.primary()`? There was a `null` check before 🤔

Reply:
good catch! Checked the code again, we should have that `null` check to keep safe.

Reply:
Could you add it to the legacy code as well 🙏? It's fine if you do it as part of this PR 👍

Reply:
see above, I have included the null pointer protection into the legacy code.